cms/navigation/wagtail_hooks.py

```diff
@@ -0,0 +1,33 @@
+from typing import TYPE_CHECKING
+
+from wagtail.permissions import ModelPermissionPolicy
+from wagtail.snippets.models import register_snippet
+from wagtail.snippets.views.snippets import SnippetViewSet
+
+from .models import MainMenu
+
+if TYPE_CHECKING:
+    from cms.users.models import User
+
+
+class NoAddModelPermissionPolicy(ModelPermissionPolicy):
+    """Model permission that doesn't allow creating more than one main menu instance."""
+
+    def user_has_permission(self, user: "User", action: str) -> bool:
```

Comment:
needs a couple of tests (both can be with a superuser):
1. main menu add view can be accessed, if a main menu doesn't exist
2. you get redirected to the Wagtail dashboard if you try to the main menu add url when a main menu exists.

The URLs:
- main menu add url would be `reverse("wagtailsnippets_navigation_mainmenu:add")`
- Wagtail dashboard `reverse("wagtailadmin_home")`


as a separate note, snippet URLs follow the 

`wagtailsnippets_{snippet._meta.app_label}_{snippet._meta.model_name}` pattern for the prefix, then `:add`, `:edit`, `:delete` for the suffix when used with `reverse()`

Reply:
Main menu viewset tests added now.